Source/Core/Core/WC24PatchEngine.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+// WC24PatchEngine
+// Allows for replacing URLs used in WC24 requests
+
+#include <algorithm>
+#include <array>
+#include <fmt/format.h>
+
+#include "Common/StringUtil.h"
+
+#include "Core/CheatCodes.h"
+#include "Core/CommonTitles.h"
+#include "Core/Config/MainSettings.h"
+#include "Core/ConfigManager.h"
+#include "Core/WC24PatchEngine.h"
+
+namespace WC24PatchEngine
+{
+static std::array<u64, 12> s_wc24_channels{
+    // Nintendo Channel
+    Titles::NINTENDO_CHANNEL_NTSC_U,
+    Titles::NINTENDO_CHANNEL_NTSC_J,
+    Titles::NINTENDO_CHANNEL_PAL,
+    Titles::FORECAST_CHANNEL_NTSC_U,
+    Titles::FORECAST_CHANNEL_NTSC_J,
+    Titles::FORECAST_CHANNEL_PAL,
+    Titles::NEWS_CHANNEL_NTSC_U,
+    Titles::NEWS_CHANNEL_NTSC_J,
+    Titles::NEWS_CHANNEL_PAL,
+    Titles::EVERYBODY_VOTES_CHANNEL_NTSC_U,
+    Titles::EVERYBODY_VOTES_CHANNEL_NTSC_J,
+    Titles::EVERYBODY_VOTES_CHANNEL_PAL,
+};
+
+static std::vector<NetworkPatch> s_patches;
+
+bool DeserializeLine(const std::string& line, NetworkPatch* patch)
+{
+  const std::vector<std::string> items = SplitString(line, ':');
+
+  if (items.size() != 3)
+    return false;
+
+  patch->source = items[0];
+  patch->replacement = items[1];
+
+  if (!TryParse(items[2], &patch->is_kd))
+    return false;
+
+  return patch;
+}
+
+void LoadPatchSection(const Common::IniFile& ini)
+{
+  std::vector<std::string> lines;
+  NetworkPatch patch;
+  ini.GetLines("WC24Patch", &lines);
+
+  for (std::string& line : lines)
+  {
+    if (line.empty())
+      continue;
+
+    if (line[0] == '$')
+    {
+      patch.name = line.substr(1, line.size() - 1);
+    }
+    else
+    {
+      if (DeserializeLine(line, &patch))
+      {
+        s_patches.push_back(patch);
+      }
+    }
+  }
+
+  ReadEnabledAndDisabled(ini, "WC24Patch", &s_patches);
```

Comment:
Does this actually do anything? I see no `WC24Patch_Enabled` section in the ini, and no way to enable or disable the individual patches either.

Reply:
Thank you so much for this catch, I must have deleted the `WC24Patch_Enabled` while doing something. It is re-added now.